### PR TITLE
Update packer-provisioner-goss.go

### PR DESCRIPTION
**Pull request by vamegh**
Custom Gossfile suport (the --gossfile flag)
Tests now supports uploading directories - ignore support disabled (for now)
Username and Password for auth to download goss from custom authed locations
Ssl cert check disable option - for internal services using self-signed certs

	modified:   README.md
	modified:   packer-provisioner-goss.go

### CODE REVIEW

**Comment by vamegh** [comment]
Hi @fishnix, 

Sorry about this, this should be much cleaner.

**Comment by fishnix** [comment]
Thanks @vamegh, I'll try to have a look today and get it merged up.  Cheers!

File: packer-provisioner-goss.go [hunk 1]
```diff
@@ -144,6 +159,12 @@ func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
 			if err := p.uploadFile(ui, comm, dst, src); err != nil {
 				return fmt.Errorf("Error uploading goss test: %s", err)
 			}
+		} else if s.Mode().IsDir() == true {
```
Comment (fishnix) [inline packer-provisioner-goss.go]: `} else if s.Mode().IsDir() {`

also, you can fix the one above it too :)

File: packer-provisioner-goss.go [hunk 2]
```diff
@@ -211,6 +234,49 @@ func (p *Provisioner) debug() string {
 	return ""
 }
 
+func (p *Provisioner) sslFlag(cmdType string) string {
+	if p.config.SkipSSLChk == true {
```
Comment (fishnix) [inline packer-provisioner-goss.go]: `if p.config.SkipSSLChk {`

File: packer-provisioner-goss.go [hunk 3]
```diff
@@ -211,6 +234,49 @@ func (p *Provisioner) debug() string {
...
+		default:
+			return ""
+		}
+	}
+	return ""
+}
+
+// enable sudo if required
+func (p *Provisioner) enableSudo() string {
+	if p.config.UseSudo == true {
```
Comment (fishnix) [inline packer-provisioner-goss.go]: `if p.config.UseSudo {`

**Comment by vamegh** [comment]
Hi @fishnix should be fixed now, I've cleaned up all of the equal to true comparisons.

**Comment by michaelwittig** [comment]
Just for information: I merged the PR and created a new release here: https://github.com/widdix/packer-provisioner-goss/releases/tag/v0.1.1

I would prefer if we could have the release in this project instead of a fork.

**Comment by fishnix** [comment]
Thanks for the ping @michaelwittig I didn't notice the update

**Pull request merged**